src/mockS3.ts

```diff
@@ -1,13 +1,68 @@
-import AWSMock from "mock-aws-s3";
 import { S3 } from "aws-sdk";
 import temp from "temp";
+import { createWriteStream, mkdirSync, promises as fsPromises } from "fs";
+import path from "path";
+import { Stream } from "stream";
+import { promises } from "dns";
 
-export const mock = (): S3 => {
-  const dir = temp.mkdirSync();
-  console.log(`Creating temporary directory at ${dir}`);
-  process.on("beforeExit", () => {
-    console.log(`Temporary directory created at ${dir}`);
-  });
-  AWSMock.config.basePath = dir;
-  return new AWSMock.S3();
+const isStream = (candidate: S3.Body): candidate is Stream => {
+  if (candidate == null) return false;
```

Comment:
Intentional `==`?

Reply:
it'll stop `undefined` and other falsys, but I'm not sure it's 100% needed.